src/vibratobracket.ts

```diff
@@ -37,37 +61,40 @@ export class VibratoBracket extends Element {
       wave_height: 6,
       wave_width: 4,
       wave_girth: 2,
+      vibrato_width: 0,
     };
   }
 
   // Set line position of the vibrato bracket
-  setLine(line) {
+  setLine(line: number): this {
     this.line = line;
     return this;
   }
-  setHarsh(harsh) {
+  setHarsh(harsh: boolean): this {
     this.render_options.harsh = harsh;
     return this;
   }
 
   // Draw the vibrato bracket on the rendering context
-  draw() {
-    const ctx = this.context;
+  draw(): void {
+    const ctx = this.checkContext();
     this.setRendered();
+    const startStave = this.start ? this.start.getStave() : undefined;
+    const stopStave = this.stop ? this.stop.getStave() : undefined;
 
-    const y = this.start
-      ? this.start.getStave().getYForTopText(this.line)
-      : this.stop.getStave().getYForTopText(this.line);
+    const y = startStave ? startStave.getYForTopText(this.line) : stopStave ? stopStave.getYForTopText(this.line) : 0;
 
     // If start note is not set then vibrato will be drawn
     // from the beginning of the stave
-    const start_x = this.start ? this.start.getAbsoluteX() : this.stop.getStave().getTieStartX();
+    const start_x = this.start ? this.start.getAbsoluteX() : stopStave ? stopStave.getTieStartX() : 0;
```

Comment:
This seems to change the logic to silently fail. If `this.start` is not set, then `stopStave` must be set.

Reply:
changed to assertion